[Feature] Support broker load without broker

xiaoyong-z opened the pull request:
Signed-off-by: xyz <a997647204@gmail.com>

## What type of PR is this：
- [ ] bug
- [x] feature
- [ ] enhancement
- [ ] refactor
- [ ] others

## Which issues of this PR fixes ：
<!--
Usage: `Fixes #<issue number>`, or `Fixes (paste link of issue)`.
_If PR is about `failing-tests or flakes`, please post the related issues/tests in a comment and do not use `Fixes`_*
-->
Fixes #9048 

## Problem Summary(Required) ：
<!-- (Please describe the changes you have made. In which scenarios will this bug be triggered and what measures have you taken to fix the bug?) -->

xiaoyong-z committed:
[Feature] support broker load without broker

Signed-off-by: xyz <a997647204@gmail.com>

xiaoyong-z committed:
disable ssl by default

Signed-off-by: xyz <a997647204@gmail.com>

xiaoyong-z commented:
run starrocks_fe_unittest

wanpengfei-git commented:
#### [FE PR Coverage Check]

:disappointed: **fail** : 66 / 863 (07.65%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/fs/hdfs/HdfsFs.java|0|35|00.00%|[30, 39, 40, 41, 42, 43, 44, 45, 46, 49, 50, 51, 54, 55, 56, 59, 61, 63, 64, 65, 67, 71, 73, 76, 77, 81, 82, 86, 87, 90, 94, 98, 99, 101, 106]|
|:large_blue_circle:|com/starrocks/fs/hdfs/WildcardURI.java|0|15|00.00%|[42, 46, 52, 53, 54, 55, 56, 57, 58, 59, 60, 61, 64, 68, 72]|
|:large_blue_circle:|com/starrocks/fs/hdfs/HdfsFsIdentity.java|0|28|00.00%|[25, 26, 27, 28, 32, 33, 34, 35, 36, 37, 42, 43, 45, 46, 48, 49, 51, 52, 53, 54, 56, 57, 59, 60, 61, 63, 64, 66]|
|:large_blue_circle:|com/starrocks/fs/hdfs/HdfsFsManager.java|15|524|02.86%|[61, 62, 65, 66, 69, 70, 71, 72, 73, 74, 75, 76, 82, 83, 89, 90, 96, 97, 103, 104, 107, 108, 110, 112, 114, 115, 120, 121, 124, 125, 128, 214, 215, 219, 220, 234, 235, 236, 237, 239, 240, 241, 242, 243, 244, 245, 246, 247, 248, 249, 250, 251, 253, 255, 280, 281, 282, 283, 284, 285, 287, 288, 291, 292, 293, 294, 296, 297, 299, 303, 304, 306, 310, 311, 312, 313, 314, 315, 316, 318, 320, 322, 325, 327, 328, 330, 332, 333, 334, 335, 336, 338, 339, 340, 343, 344, 345, 346, 348, 367, 368, 369, 370, 371, 372, 374, 375, 376, 378, 379, 380, 382, 383, 384, 389, 390, 391, 392, 393, 394, 395, 397, 399, 401, 404, 406, 407, 409, 410, 411, 412, 413, 414, 415, 416, 417, 418, 419, 421, 422, 423, 426, 427, 428, 429, 431, 450, 451, 452, 453, 454, 455, 459, 460, 461, 463, 464, 465, 467, 468, 469, 471, 472, 473, 474, 475, 476, 477, 479, 481, 483, 486, 488, 489, 491, 492, 493, 494, 495, 496, 497, 498, 499, 500, 501, 502, 504, 505, 506, 509, 510, 511, 512, 514, 533, 534, 535, 536, 537, 538, 539, 540, 541, 543, 544, 545, 547, 548, 549, 554, 555, 557, 558, 559, 561, 563, 565, 567, 570, 573, 574, 576, 577, 578, 579, 580, 581, 582, 583, 584, 585, 586, 587, 589, 590, 591, 594, 595, 596, 597, 599, 618, 619, 620, 621, 622, 623, 624, 625, 626, 628, 629, 630, 632, 633, 634, 639, 640, 641, 642, 643, 644, 645, 647, 649, 651, 654, 656, 657, 659, 660, 661, 662, 663, 664, 665, 666, 667, 668, 669, 670, 671, 673, 674, 675, 678, 679, 680, 681, 683, 695, 696, 697, 698, 699, 700, 701, 702, 703, 705, 706, 707, 709, 710, 711, 716, 717, 718, 719, 720, 721, 722, 724, 726, 728, 731, 733, 734, 736, 737, 738, 739, 740, 741, 742, 743, 744, 745, 746, 747, 749, 750, 751, 754, 755, 756, 757, 759, 765, 766, 771, 772, 773, 774, 776, 777, 778, 779, 781, 782, 783, 784, 785, 786, 787, 789, 790, 792, 794, 797, 799, 801, 802, 803, 804, 805, 806, 807, 808, 809, 813, 814, 815, 817, 818, 819, 820, 821, 822, 823, 826, 827, 828, 829, 832, 833, 834, 836, 837, 838, 840, 841, 842, 843, 844, 845, 848, 849, 850, 852, 853, 854, 855, 856, 857, 862, 863, 864, 866, 867, 868, 869, 870, 871, 872, 873, 874, 875, 880, 881, 884, 885, 886, 887, 888, 889, 891, 895, 896, 897, 899, 902, 903, 905, 908, 909, 910, 912, 913, 916, 917, 919, 920, 921, 923, 924, 925, 931, 935, 936, 938, 939, 940, 941, 943, 945, 946, 949, 950, 951, 953, 955, 956, 957, 958, 959, 960, 961, 962, 967, 968, 969, 970, 971, 975, 976, 977, 978, 979, 980, 981, 984, 985, 987, 988, 989, 990, 991, 993, 995, 996, 999, 1003, 1004, 1005, 1006, 1007, 1009, 1010, 1011, 1019, 1020, 1021, 1023, 1024, 1025, 1027, 1028, 1031]|
|:large_blue_circle:|com/starrocks/fs/hdfs/HdfsFsStreamManager.java|5|51|09.80%|[41, 42, 46, 47, 50, 51, 52, 54, 58, 59, 60, 62, 66, 68, 69, 70, 72, 73, 74, 75, 78, 80, 81, 82, 84, 85, 86, 87, 94, 95, 96, 97, 98, 101, 102, 106, 107, 115, 116, 117, 118, 119, 122, 123, 127, 128]|
|:large_blue_circle:|com/starrocks/fs/hdfs/HdfsService.java|4|44|09.09%|[53, 54, 58, 59, 60, 61, 63, 66, 67, 68, 70, 71, 72, 76, 77, 78, 82, 83, 84, 88, 89, 94, 95, 101, 102, 107, 108, 109, 113, 114, 115, 119, 120, 121, 126, 127, 128, 132, 133, 134]|
|:large_blue_circle:|com/starrocks/fs/HdfsUtil.java|23|127|18.11%|[55, 64, 65, 77, 78, 79, 80, 89, 97, 100, 105, 110, 119, 134, 137, 138, 139, 140, 141, 142, 145, 146, 149, 150, 151, 153, 155, 158, 159, 162, 163, 164, 165, 179, 181, 182, 183, 185, 187, 199, 200, 201, 202, 204, 205, 206, 208, 209, 210, 213, 214, 215, 216, 217, 218, 219, 220, 223, 225, 226, 228, 229, 231, 232, 233, 235, 245, 246, 247, 248, 251, 253, 254, 258, 259, 263, 265, 266, 267, 276, 277, 278, 279, 280, 283, 285, 286, 287, 288, 289, 292, 293, 297, 299, 300, 301, 305, 306, 309, 310, 311, 312, 314, 315]|
|:large_blue_circle:|com/starrocks/analysis/BrokerDesc.java|2|9|22.22%|[56, 57, 58, 59, 60, 61, 63]|
|:large_blue_circle:|com/starrocks/planner/FileScanNode.java|12|24|50.00%|[240, 241, 243, 244, 245, 389, 390, 394, 428, 429, 431, 670]|
|:large_blue_circle:|com/starrocks/load/loadv2/BrokerLoadPendingTask.java|2|3|66.67%|[82]|
|:large_blue_circle:|com/starrocks/common/Config.java|3|3|100.00%|[]|

xiaoyong-z committed:
add some test

Signed-off-by: xyz <a997647204@gmail.com>

wanpengfei-git commented:
#### [FE PR Coverage Check]

:disappointed: **fail** : 179 / 863 (20.74%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/fs/hdfs/HdfsFsStreamManager.java|5|51|09.80%|[41, 42, 46, 47, 50, 51, 52, 54, 58, 59, 60, 62, 66, 68, 69, 70, 72, 73, 74, 75, 78, 80, 81, 82, 84, 85, 86, 87, 94, 95, 96, 97, 98, 101, 102, 106, 107, 115, 116, 117, 118, 119, 122, 123, 127, 128]|
|:large_blue_circle:|com/starrocks/fs/hdfs/HdfsService.java|4|44|09.09%|[53, 54, 58, 59, 60, 61, 63, 66, 67, 68, 70, 71, 72, 76, 77, 78, 82, 83, 84, 88, 89, 94, 95, 101, 102, 107, 108, 109, 113, 114, 115, 119, 120, 121, 126, 127, 128, 132, 133, 134]|
|:large_blue_circle:|com/starrocks/fs/hdfs/HdfsFsManager.java|89|524|16.98%|[65, 66, 69, 70, 71, 72, 73, 74, 75, 76, 82, 83, 89, 90, 96, 97, 103, 104, 107, 108, 110, 112, 114, 115, 124, 125, 128, 214, 215, 219, 220, 237, 244, 245, 246, 247, 248, 249, 250, 251, 253, 283, 284, 285, 287, 288, 297, 299, 304, 306, 318, 325, 336, 339, 340, 344, 345, 346, 375, 376, 379, 380, 383, 384, 397, 404, 419, 422, 423, 427, 428, 429, 450, 451, 452, 453, 454, 455, 459, 460, 461, 463, 464, 465, 467, 468, 469, 471, 472, 473, 474, 475, 476, 477, 479, 481, 483, 486, 488, 489, 491, 492, 493, 494, 495, 496, 497, 498, 499, 500, 501, 502, 504, 505, 506, 509, 510, 511, 512, 514, 533, 534, 535, 536, 537, 538, 539, 540, 541, 543, 544, 545, 547, 548, 549, 554, 555, 557, 558, 559, 561, 563, 565, 567, 570, 573, 574, 576, 577, 578, 579, 580, 581, 582, 583, 584, 585, 586, 587, 589, 590, 591, 594, 595, 596, 597, 599, 618, 619, 620, 621, 622, 623, 624, 625, 626, 628, 629, 630, 632, 633, 634, 639, 640, 641, 642, 643, 644, 645, 647, 649, 651, 654, 656, 657, 659, 660, 661, 662, 663, 664, 665, 666, 667, 668, 669, 670, 671, 673, 674, 675, 678, 679, 680, 681, 683, 695, 696, 697, 698, 699, 700, 701, 702, 703, 705, 706, 707, 709, 710, 711, 716, 717, 718, 719, 720, 721, 722, 724, 726, 728, 731, 733, 734, 736, 737, 738, 739, 740, 741, 742, 743, 744, 745, 746, 747, 749, 750, 751, 754, 755, 756, 757, 759, 765, 766, 771, 772, 773, 774, 776, 777, 778, 779, 781, 782, 783, 784, 785, 786, 787, 789, 790, 792, 794, 797, 799, 801, 802, 803, 804, 805, 806, 807, 808, 809, 813, 814, 815, 817, 818, 819, 820, 821, 822, 823, 826, 827, 828, 829, 832, 833, 834, 836, 837, 838, 840, 841, 842, 843, 844, 845, 848, 849, 850, 852, 853, 854, 855, 856, 857, 862, 863, 864, 866, 867, 868, 869, 870, 871, 872, 873, 874, 875, 880, 881, 884, 885, 886, 887, 888, 889, 891, 895, 896, 897, 899, 902, 903, 905, 908, 909, 910, 912, 913, 916, 917, 919, 920, 921, 923, 924, 925, 931, 935, 936, 938, 939, 940, 941, 943, 945, 946, 949, 950, 951, 953, 955, 956, 957, 958, 959, 960, 961, 962, 967, 968, 969, 970, 971, 975, 976, 977, 978, 979, 980, 981, 984, 985, 987, 988, 989, 990, 991, 993, 995, 996, 999, 1003, 1004, 1005, 1006, 1007, 1009, 1010, 1011, 1019, 1020, 1021, 1023, 1024, 1025, 1027, 1028, 1031]|
|:large_blue_circle:|com/starrocks/fs/HdfsUtil.java|23|127|18.11%|[55, 64, 65, 77, 78, 79, 80, 89, 97, 100, 105, 110, 119, 134, 137, 138, 139, 140, 141, 142, 145, 146, 149, 150, 151, 153, 155, 158, 159, 162, 163, 164, 165, 179, 181, 182, 183, 185, 187, 199, 200, 201, 202, 204, 205, 206, 208, 209, 210, 213, 214, 215, 216, 217, 218, 219, 220, 223, 225, 226, 228, 229, 231, 232, 233, 235, 245, 246, 247, 248, 251, 253, 254, 258, 259, 263, 265, 266, 267, 276, 277, 278, 279, 280, 283, 285, 286, 287, 288, 289, 292, 293, 297, 299, 300, 301, 305, 306, 309, 310, 311, 312, 314, 315]|
|:large_blue_circle:|com/starrocks/analysis/BrokerDesc.java|2|9|22.22%|[56, 57, 58, 59, 60, 61, 63]|
|:large_blue_circle:|com/starrocks/fs/hdfs/HdfsFsIdentity.java|10|28|35.71%|[42, 43, 45, 46, 48, 49, 51, 52, 53, 54, 56, 57, 59, 60, 61, 63, 64, 66]|
|:large_blue_circle:|com/starrocks/planner/FileScanNode.java|12|24|50.00%|[240, 241, 243, 244, 245, 389, 390, 394, 428, 429, 431, 670]|
|:large_blue_circle:|com/starrocks/fs/hdfs/HdfsFs.java|18|35|51.43%|[59, 61, 63, 64, 65, 67, 71, 73, 81, 82, 86, 87, 90, 98, 99, 101, 106]|
|:large_blue_circle:|com/starrocks/load/loadv2/BrokerLoadPendingTask.java|2|3|66.67%|[82]|
|:large_blue_circle:|com/starrocks/fs/hdfs/WildcardURI.java|11|15|73.33%|[56, 57, 58, 59]|
|:large_blue_circle:|com/starrocks/common/Config.java|3|3|100.00%|[]|

xiaoyong-z commented:
run starrocks_admit_test

xiaoyong-z commented on fe/fe-core/src/main/java/com/starrocks/planner/FileScanNode.java:
```diff
@@ -502,7 +524,7 @@ private void processFileGroup(
             TBrokerFileStatus fileStatus = fileStatuses.get(i);
             TFileFormatType formatType = formatType(context.fileGroup.getFileFormat(), fileStatus.path);
-            List<String> columnsFromPath = BrokerUtil.parseColumnsFromPath(fileStatus.path,
+            List<String> columnsFromPath = HdfsUtil.parseColumnsFromPath(fileStatus.path,
```
Because parseColumnsFromPath doesn't need to communicate with broker / hdfs / s3, it only parses the columns from the path. 
The codes in BrokerUtil and HdfsUtil are same.

xiaoyong-z committed:
small change

Signed-off-by: xyz <a997647204@gmail.com>

xiaoyong-z commented:
run starrocks_fe_unittest

wanpengfei-git commented:
#### [FE PR Coverage Check]

:disappointed: **fail** : 179 / 863 (20.74%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/fs/hdfs/HdfsFsStreamManager.java|5|51|09.80%|[41, 42, 46, 47, 50, 51, 52, 54, 58, 59, 60, 62, 66, 68, 69, 70, 72, 73, 74, 75, 78, 80, 81, 82, 84, 85, 86, 87, 94, 95, 96, 97, 98, 101, 102, 106, 107, 115, 116, 117, 118, 119, 122, 123, 127, 128]|
|:large_blue_circle:|com/starrocks/fs/hdfs/HdfsService.java|4|44|09.09%|[53, 54, 58, 59, 60, 61, 63, 66, 67, 68, 70, 71, 72, 76, 77, 78, 82, 83, 84, 88, 89, 94, 95, 101, 102, 107, 108, 109, 113, 114, 115, 119, 120, 121, 126, 127, 128, 132, 133, 134]|
|:large_blue_circle:|com/starrocks/fs/hdfs/HdfsFsManager.java|89|524|16.98%|[65, 66, 69, 70, 71, 72, 73, 74, 75, 76, 82, 83, 89, 90, 96, 97, 103, 104, 107, 108, 110, 112, 114, 115, 124, 125, 128, 214, 215, 219, 220, 237, 244, 245, 246, 247, 248, 249, 250, 251, 253, 283, 284, 285, 287, 288, 297, 299, 304, 306, 318, 325, 336, 339, 340, 344, 345, 346, 375, 376, 379, 380, 383, 384, 397, 404, 419, 422, 423, 427, 428, 429, 450, 451, 452, 453, 454, 455, 459, 460, 461, 463, 464, 465, 467, 468, 469, 471, 472, 473, 474, 475, 476, 477, 479, 481, 483, 486, 488, 489, 491, 492, 493, 494, 495, 496, 497, 498, 499, 500, 501, 502, 504, 505, 506, 509, 510, 511, 512, 514, 533, 534, 535, 536, 537, 538, 539, 540, 541, 543, 544, 545, 547, 548, 549, 554, 555, 557, 558, 559, 561, 563, 565, 567, 570, 573, 574, 576, 577, 578, 579, 580, 581, 582, 583, 584, 585, 586, 587, 589, 590, 591, 594, 595, 596, 597, 599, 618, 619, 620, 621, 622, 623, 624, 625, 626, 628, 629, 630, 632, 633, 634, 639, 640, 641, 642, 643, 644, 645, 647, 649, 651, 654, 656, 657, 659, 660, 661, 662, 663, 664, 665, 666, 667, 668, 669, 670, 671, 673, 674, 675, 678, 679, 680, 681, 683, 695, 696, 697, 698, 699, 700, 701, 702, 703, 705, 706, 707, 709, 710, 711, 716, 717, 718, 719, 720, 721, 722, 724, 726, 728, 731, 733, 734, 736, 737, 738, 739, 740, 741, 742, 743, 744, 745, 746, 747, 749, 750, 751, 754, 755, 756, 757, 759, 765, 766, 771, 772, 773, 774, 776, 777, 778, 779, 781, 782, 783, 784, 785, 786, 787, 789, 790, 792, 794, 797, 799, 801, 802, 803, 804, 805, 806, 807, 808, 809, 813, 814, 815, 817, 818, 819, 820, 821, 822, 823, 826, 827, 828, 829, 832, 833, 834, 836, 837, 838, 840, 841, 842, 843, 844, 845, 848, 849, 850, 852, 853, 854, 855, 856, 857, 862, 863, 864, 866, 867, 868, 869, 870, 871, 872, 873, 874, 875, 880, 881, 884, 885, 886, 887, 888, 889, 891, 895, 896, 897, 899, 902, 903, 905, 908, 909, 910, 912, 913, 916, 917, 919, 920, 921, 923, 924, 925, 931, 935, 936, 938, 939, 940, 941, 943, 945, 946, 949, 950, 951, 953, 955, 956, 957, 958, 959, 960, 961, 962, 967, 968, 969, 970, 971, 975, 976, 977, 978, 979, 980, 981, 984, 985, 987, 988, 989, 990, 991, 993, 995, 996, 999, 1003, 1004, 1005, 1006, 1007, 1009, 1010, 1011, 1019, 1020, 1021, 1023, 1024, 1025, 1027, 1028, 1031]|
|:large_blue_circle:|com/starrocks/fs/HdfsUtil.java|23|127|18.11%|[55, 64, 65, 77, 78, 79, 80, 89, 97, 100, 105, 110, 119, 134, 137, 138, 139, 140, 141, 142, 145, 146, 149, 150, 151, 153, 155, 158, 159, 162, 163, 164, 165, 179, 181, 182, 183, 185, 187, 199, 200, 201, 202, 204, 205, 206, 208, 209, 210, 213, 214, 215, 216, 217, 218, 219, 220, 223, 225, 226, 228, 229, 231, 232, 233, 235, 245, 246, 247, 248, 251, 253, 254, 258, 259, 263, 265, 266, 267, 276, 277, 278, 279, 280, 283, 285, 286, 287, 288, 289, 292, 293, 297, 299, 300, 301, 305, 306, 309, 310, 311, 312, 314, 315]|
|:large_blue_circle:|com/starrocks/analysis/BrokerDesc.java|2|9|22.22%|[56, 57, 58, 59, 60, 61, 63]|
|:large_blue_circle:|com/starrocks/fs/hdfs/HdfsFsIdentity.java|10|28|35.71%|[42, 43, 45, 46, 48, 49, 51, 52, 53, 54, 56, 57, 59, 60, 61, 63, 64, 66]|
|:large_blue_circle:|com/starrocks/planner/FileScanNode.java|12|24|50.00%|[240, 241, 243, 244, 245, 389, 390, 394, 428, 429, 431, 670]|
|:large_blue_circle:|com/starrocks/fs/hdfs/HdfsFs.java|18|35|51.43%|[59, 61, 63, 64, 65, 67, 71, 73, 81, 82, 86, 87, 90, 98, 99, 101, 106]|
|:large_blue_circle:|com/starrocks/load/loadv2/BrokerLoadPendingTask.java|2|3|66.67%|[82]|
|:large_blue_circle:|com/starrocks/fs/hdfs/WildcardURI.java|11|15|73.33%|[56, 57, 58, 59]|
|:large_blue_circle:|com/starrocks/common/Config.java|3|3|100.00%|[]|

chaoyli commented on be/src/fs/fs_hdfs.cpp:
```diff
@@ -194,8 +192,50 @@ class HdfsFileSystem : public FileSystem {
+    HdfsFsHandle handle;
+    RETURN_IF_ERROR(HdfsFsCache::instance()->get_connection(namenode, &handle, _options));
+    if (handle.type != HdfsFsHandle::Type::HDFS) {
+        return Status::InvalidArgument("invalid hdfs path");
```
Add the path info in the error status

chaoyli commented on be/src/fs/fs_hdfs.cpp:
```diff
@@ -205,20 +245,25 @@ StatusOr<std::unique_ptr<RandomAccessFile>> HdfsFileSystem::new_random_access_fi
     HdfsFsHandle handle;
-    RETURN_IF_ERROR(HdfsFsCache::instance()->get_connection(namenode, &handle));
+    RETURN_IF_ERROR(HdfsFsCache::instance()->get_connection(namenode, &handle, _options));
     if (handle.type != HdfsFsHandle::Type::HDFS) {
```
Add the path info in the error status

chaoyli commented on be/src/fs/fs_hdfs.cpp:
```diff
@@ -205,20 +245,25 @@ StatusOr<std::unique_ptr<RandomAccessFile>> HdfsFileSystem::new_random_access_fi
 
-std::unique_ptr<FileSystem> new_fs_hdfs() {
-    return std::make_unique<HdfsFileSystem>();
+std::unique_ptr<FileSystem> new_fs_hdfs(FSOptions& options) {
```
const FSOptions&

chaoyli commented on be/src/fs/fs_s3.cpp:
```diff
@@ -535,8 +567,8 @@ Status S3FileSystem::delete_dir_recursive(const std::string& dirname) {
 
-std::unique_ptr<FileSystem> new_fs_s3() {
-    return std::make_unique<S3FileSystem>();
+std::unique_ptr<FileSystem> new_fs_s3(FSOptions& options) {
```
const FSOptions&

chaoyli commented on be/src/fs/fs_s3.h:
```diff
@@ -6,6 +6,6 @@
 namespace starrocks {
 
-std::unique_ptr<FileSystem> new_fs_s3();
+std::unique_ptr<FileSystem> new_fs_s3(FSOptions& options);
```
const FSOptions&

chaoyli commented on fe/fe-core/src/main/java/com/starrocks/fs/HdfsUtil.java:
```diff
@@ -0,0 +1,317 @@
+// This file is made available under Elastic License 2.0.
```
This three line can be removed.

chaoyli commented on fe/fe-core/src/main/java/com/starrocks/fs/hdfs/HdfsFsManager.java:
```diff
@@ -0,0 +1,1038 @@
+// This file is made available under Elastic License 2.0.
+// This file is based on code available under the Apache license here:
+//   https://github.com/apache/incubator-doris/blob/master
```
The three lines can be removed

xiaoyong-z committed:
1. add const to FSOptions
2. add more infomation to error message
3. small change to file licensce header

Signed-off-by: xyz <a997647204@gmail.com>

wanpengfei-git commented:
#### [FE PR Coverage Check]

:disappointed: **fail** : 183 / 863 (21.21%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/fs/hdfs/HdfsFsStreamManager.java|5|51|09.80%|[41, 42, 46, 47, 50, 51, 52, 54, 58, 59, 60, 62, 66, 68, 69, 70, 72, 73, 74, 75, 78, 80, 81, 82, 84, 85, 86, 87, 94, 95, 96, 97, 98, 101, 102, 106, 107, 115, 116, 117, 118, 119, 122, 123, 127, 128]|
|:large_blue_circle:|com/starrocks/fs/hdfs/HdfsService.java|4|44|09.09%|[53, 54, 58, 59, 60, 61, 63, 66, 67, 68, 70, 71, 72, 76, 77, 78, 82, 83, 84, 88, 89, 94, 95, 101, 102, 107, 108, 109, 113, 114, 115, 119, 120, 121, 126, 127, 128, 132, 133, 134]|
|:large_blue_circle:|com/starrocks/fs/hdfs/HdfsFsManager.java|91|524|17.37%|[59, 60, 63, 64, 65, 66, 67, 68, 69, 70, 76, 77, 83, 84, 90, 91, 97, 98, 101, 102, 104, 106, 108, 109, 118, 119, 122, 208, 209, 213, 214, 231, 238, 239, 240, 241, 242, 243, 244, 245, 247, 277, 278, 279, 281, 282, 291, 293, 298, 300, 312, 319, 330, 333, 334, 338, 339, 340, 369, 370, 373, 374, 377, 378, 391, 398, 413, 416, 417, 421, 422, 423, 444, 445, 446, 447, 448, 449, 453, 454, 455, 457, 458, 459, 461, 462, 463, 465, 466, 467, 468, 469, 470, 471, 473, 475, 477, 480, 482, 483, 485, 486, 487, 488, 489, 490, 491, 492, 493, 494, 495, 496, 498, 499, 500, 503, 504, 505, 506, 508, 527, 528, 529, 530, 531, 532, 533, 534, 535, 537, 538, 539, 541, 542, 543, 548, 549, 551, 552, 553, 555, 557, 559, 561, 564, 567, 568, 570, 571, 572, 573, 574, 575, 576, 577, 578, 579, 580, 581, 583, 584, 585, 588, 589, 590, 591, 593, 612, 613, 614, 615, 616, 617, 618, 619, 620, 622, 623, 624, 626, 627, 628, 633, 634, 635, 636, 637, 638, 639, 641, 643, 645, 648, 650, 651, 653, 654, 655, 656, 657, 658, 659, 660, 661, 662, 663, 664, 665, 667, 668, 669, 672, 673, 674, 675, 677, 689, 690, 691, 692, 693, 694, 695, 696, 697, 699, 700, 701, 703, 704, 705, 710, 711, 712, 713, 714, 715, 716, 718, 720, 722, 725, 727, 728, 730, 731, 732, 733, 734, 735, 736, 737, 738, 739, 740, 741, 743, 744, 745, 748, 749, 750, 751, 753, 759, 760, 765, 766, 767, 768, 770, 771, 772, 773, 775, 776, 777, 778, 779, 780, 781, 783, 784, 786, 788, 791, 793, 795, 796, 797, 798, 799, 800, 801, 802, 803, 807, 808, 809, 811, 812, 813, 814, 815, 816, 817, 820, 821, 822, 823, 826, 827, 828, 830, 831, 832, 834, 835, 836, 837, 838, 839, 842, 843, 844, 846, 847, 848, 849, 850, 851, 856, 857, 858, 860, 861, 862, 863, 864, 865, 866, 867, 868, 869, 874, 875, 878, 879, 880, 881, 882, 883, 885, 889, 890, 891, 893, 896, 897, 899, 902, 903, 904, 906, 907, 910, 911, 913, 914, 915, 917, 918, 919, 925, 929, 930, 932, 933, 934, 935, 937, 939, 940, 943, 944, 945, 947, 949, 950, 951, 952, 953, 954, 955, 956, 961, 962, 963, 964, 965, 969, 970, 971, 972, 973, 974, 975, 978, 979, 981, 982, 983, 984, 985, 987, 989, 990, 993, 997, 998, 999, 1000, 1001, 1003, 1004, 1005, 1014, 1015, 1017, 1018, 1019, 1021, 1022]|
|:large_blue_circle:|com/starrocks/fs/HdfsUtil.java|23|127|18.11%|[51, 60, 61, 73, 74, 75, 76, 85, 93, 96, 101, 106, 115, 130, 133, 134, 135, 136, 137, 138, 141, 142, 145, 146, 147, 149, 151, 154, 155, 158, 159, 160, 161, 175, 177, 178, 179, 181, 183, 195, 196, 197, 198, 200, 201, 202, 204, 205, 206, 209, 210, 211, 212, 213, 214, 215, 216, 219, 221, 222, 224, 225, 227, 228, 229, 231, 241, 242, 243, 244, 247, 249, 250, 254, 255, 259, 261, 262, 263, 272, 273, 274, 275, 276, 279, 281, 282, 283, 284, 285, 288, 289, 293, 295, 296, 297, 301, 302, 305, 306, 307, 308, 310, 311]|
|:large_blue_circle:|com/starrocks/analysis/BrokerDesc.java|2|9|22.22%|[56, 57, 58, 59, 60, 61, 63]|
|:large_blue_circle:|com/starrocks/fs/hdfs/HdfsFsIdentity.java|10|28|35.71%|[42, 43, 45, 46, 48, 49, 51, 52, 53, 54, 56, 57, 59, 60, 61, 63, 64, 66]|
|:large_blue_circle:|com/starrocks/planner/FileScanNode.java|12|24|50.00%|[240, 241, 243, 244, 245, 389, 390, 394, 428, 429, 431, 670]|
|:large_blue_circle:|com/starrocks/fs/hdfs/HdfsFs.java|20|35|57.14%|[59, 61, 63, 64, 65, 67, 71, 73, 81, 82, 86, 87, 90, 99, 106]|
|:large_blue_circle:|com/starrocks/load/loadv2/BrokerLoadPendingTask.java|2|3|66.67%|[82]|
|:large_blue_circle:|com/starrocks/fs/hdfs/WildcardURI.java|11|15|73.33%|[56, 57, 58, 59]|
|:large_blue_circle:|com/starrocks/common/Config.java|3|3|100.00%|[]|

xiaoyong-z commented:
run starrocks_be_unittest

xiaoyong-z commented:
run starrocks_fe_unittest

xiaoyong-z commented:
run starrocks_be_unittest

wanpengfei-git commented:
#### [FE PR Coverage Check]

:disappointed: **fail** : 183 / 863 (21.21%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/fs/hdfs/HdfsFsStreamManager.java|5|51|09.80%|[41, 42, 46, 47, 50, 51, 52, 54, 58, 59, 60, 62, 66, 68, 69, 70, 72, 73, 74, 75, 78, 80, 81, 82, 84, 85, 86, 87, 94, 95, 96, 97, 98, 101, 102, 106, 107, 115, 116, 117, 118, 119, 122, 123, 127, 128]|
|:large_blue_circle:|com/starrocks/fs/hdfs/HdfsService.java|4|44|09.09%|[53, 54, 58, 59, 60, 61, 63, 66, 67, 68, 70, 71, 72, 76, 77, 78, 82, 83, 84, 88, 89, 94, 95, 101, 102, 107, 108, 109, 113, 114, 115, 119, 120, 121, 126, 127, 128, 132, 133, 134]|
|:large_blue_circle:|com/starrocks/fs/hdfs/HdfsFsManager.java|91|524|17.37%|[59, 60, 63, 64, 65, 66, 67, 68, 69, 70, 76, 77, 83, 84, 90, 91, 97, 98, 101, 102, 104, 106, 108, 109, 118, 119, 122, 208, 209, 213, 214, 231, 238, 239, 240, 241, 242, 243, 244, 245, 247, 277, 278, 279, 281, 282, 291, 293, 298, 300, 312, 319, 330, 333, 334, 338, 339, 340, 369, 370, 373, 374, 377, 378, 391, 398, 413, 416, 417, 421, 422, 423, 444, 445, 446, 447, 448, 449, 453, 454, 455, 457, 458, 459, 461, 462, 463, 465, 466, 467, 468, 469, 470, 471, 473, 475, 477, 480, 482, 483, 485, 486, 487, 488, 489, 490, 491, 492, 493, 494, 495, 496, 498, 499, 500, 503, 504, 505, 506, 508, 527, 528, 529, 530, 531, 532, 533, 534, 535, 537, 538, 539, 541, 542, 543, 548, 549, 551, 552, 553, 555, 557, 559, 561, 564, 567, 568, 570, 571, 572, 573, 574, 575, 576, 577, 578, 579, 580, 581, 583, 584, 585, 588, 589, 590, 591, 593, 612, 613, 614, 615, 616, 617, 618, 619, 620, 622, 623, 624, 626, 627, 628, 633, 634, 635, 636, 637, 638, 639, 641, 643, 645, 648, 650, 651, 653, 654, 655, 656, 657, 658, 659, 660, 661, 662, 663, 664, 665, 667, 668, 669, 672, 673, 674, 675, 677, 689, 690, 691, 692, 693, 694, 695, 696, 697, 699, 700, 701, 703, 704, 705, 710, 711, 712, 713, 714, 715, 716, 718, 720, 722, 725, 727, 728, 730, 731, 732, 733, 734, 735, 736, 737, 738, 739, 740, 741, 743, 744, 745, 748, 749, 750, 751, 753, 759, 760, 765, 766, 767, 768, 770, 771, 772, 773, 775, 776, 777, 778, 779, 780, 781, 783, 784, 786, 788, 791, 793, 795, 796, 797, 798, 799, 800, 801, 802, 803, 807, 808, 809, 811, 812, 813, 814, 815, 816, 817, 820, 821, 822, 823, 826, 827, 828, 830, 831, 832, 834, 835, 836, 837, 838, 839, 842, 843, 844, 846, 847, 848, 849, 850, 851, 856, 857, 858, 860, 861, 862, 863, 864, 865, 866, 867, 868, 869, 874, 875, 878, 879, 880, 881, 882, 883, 885, 889, 890, 891, 893, 896, 897, 899, 902, 903, 904, 906, 907, 910, 911, 913, 914, 915, 917, 918, 919, 925, 929, 930, 932, 933, 934, 935, 937, 939, 940, 943, 944, 945, 947, 949, 950, 951, 952, 953, 954, 955, 956, 961, 962, 963, 964, 965, 969, 970, 971, 972, 973, 974, 975, 978, 979, 981, 982, 983, 984, 985, 987, 989, 990, 993, 997, 998, 999, 1000, 1001, 1003, 1004, 1005, 1014, 1015, 1017, 1018, 1019, 1021, 1022]|
|:large_blue_circle:|com/starrocks/fs/HdfsUtil.java|23|127|18.11%|[51, 60, 61, 73, 74, 75, 76, 85, 93, 96, 101, 106, 115, 130, 133, 134, 135, 136, 137, 138, 141, 142, 145, 146, 147, 149, 151, 154, 155, 158, 159, 160, 161, 175, 177, 178, 179, 181, 183, 195, 196, 197, 198, 200, 201, 202, 204, 205, 206, 209, 210, 211, 212, 213, 214, 215, 216, 219, 221, 222, 224, 225, 227, 228, 229, 231, 241, 242, 243, 244, 247, 249, 250, 254, 255, 259, 261, 262, 263, 272, 273, 274, 275, 276, 279, 281, 282, 283, 284, 285, 288, 289, 293, 295, 296, 297, 301, 302, 305, 306, 307, 308, 310, 311]|
|:large_blue_circle:|com/starrocks/analysis/BrokerDesc.java|2|9|22.22%|[56, 57, 58, 59, 60, 61, 63]|
|:large_blue_circle:|com/starrocks/fs/hdfs/HdfsFsIdentity.java|10|28|35.71%|[42, 43, 45, 46, 48, 49, 51, 52, 53, 54, 56, 57, 59, 60, 61, 63, 64, 66]|
|:large_blue_circle:|com/starrocks/planner/FileScanNode.java|12|24|50.00%|[240, 241, 243, 244, 245, 389, 390, 394, 428, 429, 431, 670]|
|:large_blue_circle:|com/starrocks/fs/hdfs/HdfsFs.java|20|35|57.14%|[59, 61, 63, 64, 65, 67, 71, 73, 81, 82, 86, 87, 90, 99, 106]|
|:large_blue_circle:|com/starrocks/load/loadv2/BrokerLoadPendingTask.java|2|3|66.67%|[82]|
|:large_blue_circle:|com/starrocks/fs/hdfs/WildcardURI.java|11|15|73.33%|[56, 57, 58, 59]|
|:large_blue_circle:|com/starrocks/common/Config.java|3|3|100.00%|[]|

wanpengfei-git commented:
run starrocks_admit_test

wanpengfei-git commented:
run starrocks_admit_test

wanpengfei-git commented:
#### [FE PR Coverage Check]

:disappointed: **fail** : 183 / 863 (21.21%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/fs/hdfs/HdfsFsStreamManager.java|5|51|09.80%|[41, 42, 46, 47, 50, 51, 52, 54, 58, 59, 60, 62, 66, 68, 69, 70, 72, 73, 74, 75, 78, 80, 81, 82, 84, 85, 86, 87, 94, 95, 96, 97, 98, 101, 102, 106, 107, 115, 116, 117, 118, 119, 122, 123, 127, 128]|
|:large_blue_circle:|com/starrocks/fs/hdfs/HdfsService.java|4|44|09.09%|[53, 54, 58, 59, 60, 61, 63, 66, 67, 68, 70, 71, 72, 76, 77, 78, 82, 83, 84, 88, 89, 94, 95, 101, 102, 107, 108, 109, 113, 114, 115, 119, 120, 121, 126, 127, 128, 132, 133, 134]|
|:large_blue_circle:|com/starrocks/fs/hdfs/HdfsFsManager.java|91|524|17.37%|[59, 60, 63, 64, 65, 66, 67, 68, 69, 70, 76, 77, 83, 84, 90, 91, 97, 98, 101, 102, 104, 106, 108, 109, 118, 119, 122, 208, 209, 213, 214, 231, 238, 239, 240, 241, 242, 243, 244, 245, 247, 277, 278, 279, 281, 282, 291, 293, 298, 300, 312, 319, 330, 333, 334, 338, 339, 340, 369, 370, 373, 374, 377, 378, 391, 398, 413, 416, 417, 421, 422, 423, 444, 445, 446, 447, 448, 449, 453, 454, 455, 457, 458, 459, 461, 462, 463, 465, 466, 467, 468, 469, 470, 471, 473, 475, 477, 480, 482, 483, 485, 486, 487, 488, 489, 490, 491, 492, 493, 494, 495, 496, 498, 499, 500, 503, 504, 505, 506, 508, 527, 528, 529, 530, 531, 532, 533, 534, 535, 537, 538, 539, 541, 542, 543, 548, 549, 551, 552, 553, 555, 557, 559, 561, 564, 567, 568, 570, 571, 572, 573, 574, 575, 576, 577, 578, 579, 580, 581, 583, 584, 585, 588, 589, 590, 591, 593, 612, 613, 614, 615, 616, 617, 618, 619, 620, 622, 623, 624, 626, 627, 628, 633, 634, 635, 636, 637, 638, 639, 641, 643, 645, 648, 650, 651, 653, 654, 655, 656, 657, 658, 659, 660, 661, 662, 663, 664, 665, 667, 668, 669, 672, 673, 674, 675, 677, 689, 690, 691, 692, 693, 694, 695, 696, 697, 699, 700, 701, 703, 704, 705, 710, 711, 712, 713, 714, 715, 716, 718, 720, 722, 725, 727, 728, 730, 731, 732, 733, 734, 735, 736, 737, 738, 739, 740, 741, 743, 744, 745, 748, 749, 750, 751, 753, 759, 760, 765, 766, 767, 768, 770, 771, 772, 773, 775, 776, 777, 778, 779, 780, 781, 783, 784, 786, 788, 791, 793, 795, 796, 797, 798, 799, 800, 801, 802, 803, 807, 808, 809, 811, 812, 813, 814, 815, 816, 817, 820, 821, 822, 823, 826, 827, 828, 830, 831, 832, 834, 835, 836, 837, 838, 839, 842, 843, 844, 846, 847, 848, 849, 850, 851, 856, 857, 858, 860, 861, 862, 863, 864, 865, 866, 867, 868, 869, 874, 875, 878, 879, 880, 881, 882, 883, 885, 889, 890, 891, 893, 896, 897, 899, 902, 903, 904, 906, 907, 910, 911, 913, 914, 915, 917, 918, 919, 925, 929, 930, 932, 933, 934, 935, 937, 939, 940, 943, 944, 945, 947, 949, 950, 951, 952, 953, 954, 955, 956, 961, 962, 963, 964, 965, 969, 970, 971, 972, 973, 974, 975, 978, 979, 981, 982, 983, 984, 985, 987, 989, 990, 993, 997, 998, 999, 1000, 1001, 1003, 1004, 1005, 1014, 1015, 1017, 1018, 1019, 1021, 1022]|
|:large_blue_circle:|com/starrocks/fs/HdfsUtil.java|23|127|18.11%|[51, 60, 61, 73, 74, 75, 76, 85, 93, 96, 101, 106, 115, 130, 133, 134, 135, 136, 137, 138, 141, 142, 145, 146, 147, 149, 151, 154, 155, 158, 159, 160, 161, 175, 177, 178, 179, 181, 183, 195, 196, 197, 198, 200, 201, 202, 204, 205, 206, 209, 210, 211, 212, 213, 214, 215, 216, 219, 221, 222, 224, 225, 227, 228, 229, 231, 241, 242, 243, 244, 247, 249, 250, 254, 255, 259, 261, 262, 263, 272, 273, 274, 275, 276, 279, 281, 282, 283, 284, 285, 288, 289, 293, 295, 296, 297, 301, 302, 305, 306, 307, 308, 310, 311]|
|:large_blue_circle:|com/starrocks/analysis/BrokerDesc.java|2|9|22.22%|[56, 57, 58, 59, 60, 61, 63]|
|:large_blue_circle:|com/starrocks/fs/hdfs/HdfsFsIdentity.java|10|28|35.71%|[42, 43, 45, 46, 48, 49, 51, 52, 53, 54, 56, 57, 59, 60, 61, 63, 64, 66]|
|:large_blue_circle:|com/starrocks/planner/FileScanNode.java|12|24|50.00%|[240, 241, 243, 244, 245, 389, 390, 394, 428, 429, 431, 670]|
|:large_blue_circle:|com/starrocks/fs/hdfs/HdfsFs.java|20|35|57.14%|[59, 61, 63, 64, 65, 67, 71, 73, 81, 82, 86, 87, 90, 99, 106]|
|:large_blue_circle:|com/starrocks/load/loadv2/BrokerLoadPendingTask.java|2|3|66.67%|[82]|
|:large_blue_circle:|com/starrocks/fs/hdfs/WildcardURI.java|11|15|73.33%|[56, 57, 58, 59]|
|:large_blue_circle:|com/starrocks/common/Config.java|3|3|100.00%|[]|

xiaoyong-z commented:
run starrocks_admit_test
